proof/refine/AARCH64/InvariantUpdates_H.thy

```diff
@@ -5,11 +5,102 @@
  *)
 
 theory InvariantUpdates_H
-imports Invariants_H
+imports ArchInvsLemmas_H
 begin
 
+(* generic consequences which require arch-specific proofs *)
+arch_requalify_facts
+  cte_wp_atE'
+  cte_wp_at_cteI'
+  tcb_at_cte_at'
+  typ_at_lift_valid_cap'
+  valid_arch_tcb_lift'
+  tcb_at_cte_at'
+  gen_objBitsT_simps
+
+(* arch-specific typ_at_lifts in Arch *)
+lemmas gen_typ_at_lifts =
+         typ_at_lift_tcb' typ_at_lift_ep' typ_at_lift_ntfn' typ_at_lift_cte' typ_at_lift_cte_at'
+         typ_at_lift_valid_untyped' typ_at_lift_valid_cap' valid_bound_tcb_lift
+         valid_arch_tcb_lift'
```

Comment:
A note for the distant future, I think that this will be a problem in `rt` with the more complicated `typ_at_lifts` setup we have there. I think for that we might want to add locales and do the arch-split dance between `Invariants_H`  and `ArchInvsLemas_H`.

Probably not worth the effort, but in a perfect world it would be nice to pull that setup back to master to test out arch-splitting it there.

Reply:
I don't quite understand. What kind of setup does rt have? If it cheers you up, `gen_typ_at_lifts` is so far unused, it's only the consequence of hiding the arch-specific ones under `typ_at_lifts` in Arch, and I left this behind as a precaution.

Reply:
Given your comment and how it's unused, should I remove it to prevent a derp later?

Reply:
In `rt` we have a couple of locales that contain all of the `typ_at'` lifting rules and that are then interpreted as needed for any function that preserves `typ_at'`. See the whole section building up to https://github.com/seL4/l4v/blob/rt/proof/refine/RISCV64/Invariants_H.thy#L2777 if you're particularly curious. When it comes time for it to be arch-split, I imagine we'll want to keep as much of that generic as possible to reduce duplication and risk of de-sync.

Reply:
> Given your comment and how it's unused, should I remove it to prevent a derp later?

Hmm, sorry, it wasn't actually specifically `gen_typ_at_lifts` that I was commenting on, more the general approach with `typ_at_lifts` being in `Arch`. I just put this here as the place where some of it is made generic again.

Reply:
In that case, they'll end up more like the `_upd` locale hierarchy? I.e. locales get declared, we do what we can in generic, then add on the rest in Arch, and then (sigh) any stragglers that need the Arch ones in the next generic theory ... hmm. I wish there was a nicer way.

Reply:
Right, comparing it to the `_upd` locales makes sense and that does give us an approach to follow when we get up to that. Hopefully it ends up being not too bad.